safesocket/safesocket.go

```diff
@@ -57,10 +57,60 @@ func tailscaledStillStarting() bool {
 	return tailscaledProcExists()
 }
 
-// Connect connects to either path (on Unix) or the provided localhost port (on Windows).
-func Connect(path string, port uint16) (net.Conn, error) {
+// A ConnectionStrategy is a plan for how to connect to tailscaled.
```

Comment:
to tailscaled, or equivalent (e.g. IPNExtension on macOS)

Reply:
donedone